ztp/siteconfig-generator/siteConfig/siteConfigHelper.go

```diff
@@ -366,25 +366,44 @@ func applyWorkloadPinningInstallConfigOverrides(clusterSpec *Clusters) (result s
 	return clusterSpec.InstallConfigOverrides, nil
 }
 
-// getDeprecationWarnings is a helper function, currently it only adds deprecation warning for CPU Partitioning.
-// Note: should be expanded if more annotations are needed.
-func getDeprecationWarnings(clusterSpec Clusters) *annotationWarning {
+// getDeprecationWarnings is a helper function, currently it only adds
+// deprecation warning for CPU Partitioning and extraManifestPath.
+func getDeprecationWarnings(clusterSpec Clusters, deprecationField ...string) *annotationWarning {
 	deprecationWarnings := NewAnnotationWarning(ZtpDeprecationWarningAnnotationPostfix)
-	const (
-		deprecationCR      = "AgentClusterInstall"
-		deprecationField   = "cpuset"
-		deprecationMessage = "cpuset will be deprecated after OCP 4.15, please use cpuPartitioningMode for OCP versions >= 4.13"
-	)
 
-	for _, node := range clusterSpec.Nodes {
-		// If a CPUSet exists on any of the nodes, the whole cluster, regardless if SNO or not, must be setup
-		// for partitioning.
-		if node.Cpuset != "" {
-			// If Node has CPUSet add deprecation warning
-			deprecationWarnings.Add(deprecationCR, deprecationField, deprecationMessage)
-			break
+	for _, field := range deprecationField {
```

Comment:
I like this idea a lot, the only thing I'm worried about with this is scalability, if we need to add more. What do you think about changing this up a bit and moving the trigger logic away from here and let each deprecation annotation decide when it should be applied. Something like this?

```go
type AnnotationMessage struct {
	CRName          string
	Field           string
	Message         string
	ShouldBeApplied func(Clusters) bool
}
...
var (
	cpuSetAnnotationDeprecation = AnnotationMessage{
		CRName:  "AgentClusterInstall",
		Field:   "cpuset",
		Message: "cpuset will be deprecated after OCP 4.15, please use cpuPartitioningMode for OCP versions >= 4.13",
		ShouldBeApplied: func(c Clusters) bool {
			for _, node := range c.Nodes {
				if node.Cpuset != "" {
					return true
				}
			}
			return false
		},
	}

	extraManifestPathAnnotationDeprecation = AnnotationMessage{
		CRName:  "ConfigMap",
		Field:   "extraManifestPath",
		Message: "extraManifestPath will be deprecated after OCP 4.15, please use ExtraManifests.SearchPaths for OCP versions >= 4.14",
		ShouldBeApplied: func(c Clusters) bool {
			if len(c.ExtraManifestPath) > 0 {
				return true
			}
			return false
		},
	}
)
...
// getDeprecationWarnings is a helper function to add deprecation warnings.
func getDeprecationWarnings(clusterSpec Clusters, deprecationField ...AnnotationMessage) *annotationWarning {
	deprecationWarnings := NewAnnotationWarning(ZtpDeprecationWarningAnnotationPostfix)

	for _, field := range deprecationField {
		if field.ShouldBeApplied != nil && field.ShouldBeApplied(clusterSpec) {
			deprecationWarnings.Add(field.CRName, field.Field, field.Message)
		}
	}
	return deprecationWarnings
}
````

Reply:
@eggfoobar I really liked your suggestion, so i changed it accordingly with this commit df8d23ecfd531c39b05ef4c003088cf08c38a31e